File: frontend/src/dto/SchedulingDTO.ts
export class SchedulingDTO{
    private id: number;
    private name: string;
    private start: string;
    private finish: string;
    private machineId: number;

    constructor(id: number, name: string, start: string, finish: string, machineId: number){
        this.id = id;
        this.name = name;
        this.start = start;
        this.finish = finish;
        this.machineId = machineId;
    }

    public setId(id: number){
        this.id = id;
    }

    public setName(name: string){
        this.name = name;
    }

    public getStart(): string{
        return this.start;
    }

    public getFinish(): string{
        return this.finish;
    }

    public setStart(start: string){
        this.start = start;
    }

    public setFinish(finish: string){
        this.finish = finish;
    }

}

File: backend/machineMicroservice/src/main/java/com/mima/machine/service/dto/SchedulingDTO.java
package com.mima.machine.service.dto;
import java.io.Serializable;
import java.util.Objects;

/**
 * A DTO for the Scheduling entity.
 */
public class SchedulingDTO implements Serializable {

    private Long id;

    private String name;

    private String start;

    private String finish;


    private Long machineId;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getStart() {
        return start;
    }

    public void setStart(String start) {
        this.start = start;
    }

    public String getFinish() {
        return finish;
    }

    public void setFinish(String finish) {
        this.finish = finish;
    }

    public Long getMachineId() {
        return machineId;
    }

    public void setMachineId(Long machineId) {
        this.machineId = machineId;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }

        SchedulingDTO schedulingDTO = (SchedulingDTO) o;
        if (schedulingDTO.getId() == null || getId() == null) {
            return false;
        }
        return Objects.equals(getId(), schedulingDTO.getId());
    }

    @Override
    public int hashCode() {
        return Objects.hashCode(getId());
    }

    @Override
    public String toString() {
        return "SchedulingDTO{" +
            "id=" + getId() +
            ", name='" + getName() + "'" +
            ", start='" + getStart() + "'" +
            ", finish='" + getFinish() + "'" +
            ", machine=" + getMachineId() +
            "}";
    }
}
